client/document/support-user.jsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+
+function supportUserFn( { user, token, authorized } ) {
+	const url = window.location.toString();
+
+	if ( url.indexOf( '?' ) > 0 ) {
+		const cleanUrl = url.substring( 0, url.indexOf( '?' ) );
+		window.history.replaceState( {}, document.title, cleanUrl );
+	}
+
+	if ( authorized ) {
+		window.sessionStorage.setItem( 'boot_support_user', JSON.stringify( { user, token } ) );
+	}
```

Comment:
If we want to make a function I'd recommend pulling this out into it's own file, and including it as a script.

Reply:
Caught up with the earlier discussion and I'd like us to use what's in https://github.com/Automattic/wp-calypso/pull/22296/files/cd4e881d7dc9983294cebb835a6ef1b9be65453c

While the JS code is in strings, it's still easy to grok, and is unlikely to have much development on. What's here currently would need to pass another security review.

Reply:
I have no strong feelings about that. I updated the branch to iterate on the approaches. I'm fine with using either _JS-as-a-string_ or the extracted method.

Reply:
@gwwar I discussed this with @samouri and we came up with a more lightweight version (97cdb1a) of the _JS-as-JS_ approach where we get rid off all the additional stuff which isn't really necessary.

Do you think this is something we can go further with or would you still suggest switching back to _JS-as-string_? It's just meant as iteration and we can go back to the string version anytime.